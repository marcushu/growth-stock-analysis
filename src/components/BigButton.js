import React from 'react';
import PropTypes from 'prop-types';
import './buttonStyles.css'

const BigButton = (props) => {
  const handleClick = () => {
    props.selectedHandler(props.caption);
  }

  return (
    <button
      className={props.selectedState ? "btnSelected" : "UnSelected"}
      onClick={handleClick}>
      {props.caption}
      <img src={props.image} alt={props.altTxt} />
    </button>
  )
}

BigButton.propTypes = {
  caption: PropTypes.string,
  image: PropTypes.string,
  selectedState: PropTypes.bool,
  selectedHandler: PropTypes.func,
  altTxt: PropTypes.string
}

export default BigButton