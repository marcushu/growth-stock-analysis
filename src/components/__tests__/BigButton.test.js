import React from 'react';
import { render, cleanup } from '@testing-library/react';
import BigButton from '../BigButton';

afterEach(cleanup);

describe("Big Button", () => {
 it("renders ok", () => {
    const { asFragment } = render(<BigButton
      caption="Quick Pre-qualify"
      image={"thumbsUp"}
      altTxt={"Quick Pre-qualify"}
      selectedState={false}
      selectedHandler={() => { }} />);

    expect(asFragment()).toMatchSnapshot();
  });

 it("renders with class set to UnSelected", () => {
    const { container } = render(<BigButton />);

    expect(container.innerHTML.includes("UnSelected")).toBeTruthy();
  });
});
