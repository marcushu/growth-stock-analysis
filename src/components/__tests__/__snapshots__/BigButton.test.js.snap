// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Big Button renders ok 1`] = `
<DocumentFragment>
  <button
    class="UnSelected"
  >
    Quick Pre-qualify
    <img
      alt="Quick Pre-qualify"
      src="thumbsUp"
    />
  </button>
</DocumentFragment>
`;
